### PR TITLE
refactor: move filter generator logic into individual files

### PR DESCRIPTION
**Pull request by nareddyt**
- Move most generation logic out of `filter_generator.go`
- Use interfaces for the filter generators to reduce code complexity
- Some file and package renaming

Testing done: All pre-existing unit tests pass without any modifications to inputs/outputs.

### CODE REVIEW

File: src/go/configgenerator/filtergen/cors.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package filtergen
```
Comment (TAOXUY) [inline src/go/configgenerator/filtergen/cors.go:1]: NIT: copyright header? 

Ditto for the remaining files.

File: src/go/configgenerator/filterconfig/preserve_default_http_binding_test.go [hunk 2]
```diff
@@ -1,108 +0,0 @@
...
-import (
-	"fmt"
-	"testing"
-
-	"github.com/GoogleCloudPlatform/esp-v2/tests/utils"
-	ahpb "google.golang.org/genproto/googleapis/api/annotations"
-	"google.golang.org/protobuf/encoding/prototext"
-)
-
-func TestPreserveDefaultHttpBinding(t *testing.T) {
```
Comment (TAOXUY) [inline src/go/configgenerator/filterconfig/preserve_default_http_binding_test.go:26]: Where is this test?

File: src/go/configgenerator/filterconfig/update_proto_descriptor_test.go [hunk 3]
```diff
@@ -1,348 +0,0 @@
...
-
-	"github.com/GoogleCloudPlatform/esp-v2/tests/utils"
-	descpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
-
-	confpb "google.golang.org/genproto/googleapis/api/serviceconfig"
-	"google.golang.org/protobuf/encoding/prototext"
-	"google.golang.org/protobuf/proto"
-)
-
-func TestUpdateProtoDescriptor(t *testing.T) {
```
Comment (TAOXUY) [inline src/go/configgenerator/filterconfig/update_proto_descriptor_test.go:29]: where is this test?

File: src/go/configgenerator/filterconfig/preserve_default_http_binding_test.go (hunk 2 above)
Comment (nareddyt) [reply src/go/configgenerator/filterconfig/preserve_default_http_binding_test.go:26]: Moved to src/go/configgenerator/filtergen/grpc_transcoder_test.go

File: src/go/configgenerator/filterconfig/update_proto_descriptor_test.go (hunk 3 above)
Comment (nareddyt) [reply src/go/configgenerator/filterconfig/update_proto_descriptor_test.go:29]: Moved to src/go/configgenerator/filtergen/grpc_transcoder_test.go

File: src/go/configgenerator/filtergen/cors.go (hunk 1 above)
Comment (nareddyt) [reply src/go/configgenerator/filtergen/cors.go:15]: Done

**Review by TAOXUY — APPROVED** [review]

**Review by shuoyang2016 — APPROVED** [review]

**Pull request merged**